packages/preferences/src/browser/preference-provider-user.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { ResourceProvider, Resource } from "@theia/core/lib/common";
+import { ILogger } from "@theia/core/lib/common";
+import URI from '@theia/core/lib/common/uri';
+import { DisposableCollection } from '@theia/core/lib/common';
+import { PreferenceProvider, PreferenceProviderClient, PreferenceProviderPriority } from '@theia/preferences-api/lib/browser';
+import * as jsoncparser from "jsonc-parser";
+import { ParseError } from "jsonc-parser";
+
+@injectable()
+export class PreferenceProviderUser implements PreferenceProvider {
+    protected preferencesCache: { [key: string]: any } = {};
+
+    protected client: PreferenceProviderClient | undefined;
+
+    protected readonly toDispose = new DisposableCollection();
+
+    protected resolveReady: () => void;
+
+    readonly ready = new Promise<void>(resolve => {
+        this.resolveReady = resolve;
+    });
+
+    protected preferenceResource: Promise<Resource>;
+    public readonly priority: PreferenceProviderPriority = PreferenceProviderPriority.USER;
+
+    constructor(
+        @inject(ResourceProvider) protected readonly provider: ResourceProvider,
+        @inject(ILogger) protected readonly logger: ILogger,
+    ) { }
+
+    async init(): Promise<void> {
+        this.preferenceResource = this.provider(new URI('user_storage:settings.json'));
```

Comment:
it would be nice to add `UserStorageUri.create` to avoid hardcoding the user storage schema here

Reply:
I had it in there for a while but I moved it after discussing with @svenefftinge [here](https://github.com/theia-ide/theia/pull/797#discussion_r151046792)
 Although now that I think about it, it could make sense to have the `create` method in `user-storage-uri` as this one is independant from the impl i.e even if we didn't use the filesystem as the user storage backend (but something cloud based for instance) clients would still need to create user storage uri the same way i.e `user_storage:resource`. WDYT? If so I can make a small PR and add this API.

Reply:
sounds fine with me